native/src/hooks/useEmbeddedOffers.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback } from 'react'
+
+import {
+  CategoryModel,
+  LoadSprungbrettJobReturn,
+  SPRUNGBRETT_OFFER,
+  loadSprungbrettJobs,
+  useLoadAsync,
+} from 'api-client'
+
+import { determineApiUrl } from '../utils/helpers'
+
+type EmbeddedOffers = {
+  sprungbrett?: Awaited<LoadSprungbrettJobReturn>
+}
+
+type EmbeddedOffersParams = {
+  category?: CategoryModel | null
```

Comment:
Imo there is no reason to connect this to a category, in theory also some other models could have embedded offers. Additionally, you could just pass an empty array as default then to avoid some of the checks below.
```suggestion
  embeddedOffers: OfferModel[]
```

Reply:
This does not work with the dependencies. This rerenders this hook over and over again, as embeddedOffers is not memoized, but considered a different value than category. Hard to explain, but this just doesn't work...